src/main/waveform/waveform.core.js

```diff
@@ -142,23 +142,13 @@ define([
       },
 
       /**
-       * Deal with window resize event over both waveform views.
+       * Deals with binding the window resize event and other waveform views resize events.
        */
       bindResize: function () {
         var that = this;
 
-        window.addEventListener("resize", function () {
-          that.ui.overview.hidden = true;
-          that.ui.zoom.hidden = true;
-
-          if (that.resizeTimeoutId) clearTimeout(that.resizeTimeoutId);
-          that.resizeTimeoutId = setTimeout(function(){
-            var w = that.ui.player.clientWidth;
-            var overviewWaveformData = that.origWaveformData.resample(w);
-            peaks.emit("resizeEndOverview", w, overviewWaveformData);
-            peaks.emit("window_resized", w, that.origWaveformData);
-          }, 500);
-        });
+        that.resizeListener = that.onResize.bind(that);
```

Comment:
You should call it `_resizeListener` since it's internal and shouldn't be touched by others.